cmd/snapshot-create.go

```diff
@@ -0,0 +1,161 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/Layr-Labs/sidecar/internal/logger"
+	"github.com/spf13/cobra"
+)
+
+var (
+	snapshotCreateToPath string
+	snapshotFidelity     string
+	s3Path               string
+	compressSnapshot     bool
+)
+
+var snapshotCreateCmd = &cobra.Command{
+	Use:   "snapshot-create",
+	Short: "Create a snapshot of the database",
+	Long: `Create a snapshot of the database.
+
+Currently available fidelity levels:
+- archive (default): includes chain data, EigenModel state, rewards, and staker-operator table data.
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Initialize logger (in a real scenario, you'd likely pass in a config)
+		l, _ := logger.NewLogger(&logger.LoggerConfig{Debug: true})
+
+		// Retrieve database connection details from global flags
+		dbHost, _ := cmd.Flags().GetString("database.host")
+		dbName, _ := cmd.Flags().GetString("database.db_name")
+		dbUser, _ := cmd.Flags().GetString("database.user")
+		dbPassword, _ := cmd.Flags().GetString("database.password")
+		dbPort, _ := cmd.Flags().GetInt("database.port")
+		schemaName, _ := cmd.Flags().GetString("database.schema_name")
+
+		// Retrieve chain and fidelity
+		chain, _ := cmd.Flags().GetString("chain")
+
+		// Validate fidelity
+		if snapshotFidelity != "archive" {
```

Comment:
Good to check invariants, but a default should be set via cobra for the corresponding flag.